pkg/catalog/routes.go

```diff
@@ -27,6 +27,47 @@ var wildCardRouteMatch trafficpolicy.HTTPRouteMatch = trafficpolicy.HTTPRouteMat
 	Methods:   []string{constants.WildcardHTTPMethod},
 }
 
+func (mc *MeshCatalog) listTrafficPoliciesForTrafficSplits(sourceNamespace string) []*trafficpolicy.OutboundTrafficPolicy {
+	outboundPoliciesFromSplits := []*trafficpolicy.OutboundTrafficPolicy{}
+
+	apexServices := mapset.NewSet()
+	for _, split := range mc.meshSpec.ListTrafficSplits() {
+		svc := service.MeshService{
+			Name:      split.Spec.Service,
+			Namespace: split.ObjectMeta.Namespace,
+		}
+
+		hostnames, err := mc.getServiceHostnames(svc, svc.Namespace == sourceNamespace)
+		if err != nil {
+			log.Error().Err(err).Msgf("Error getting service hostnames for apex service %v", svc)
+			continue
+		}
+		policy := trafficpolicy.NewOutboundTrafficPolicy(split.Spec.Service+"-"+split.ObjectMeta.Namespace, hostnames)
```

Comment:
```suggestion
		policy := trafficpolicy.NewOutboundTrafficPolicy(split.Spec.Service+"."+split.ObjectMeta.Namespace, hostnames)
```

With this PR : https://github.com/openservicemesh/osm/pull/2493 we have changed the naming to be `<service>.<namespace>`

Reply:
Further I feel it would be good to generalize this method  : https://github.com/openservicemesh/osm/blob/59bf80863a9a75d5070ae85c5bab8963835f71e6/pkg/catalog/ingress.go#L103  and use it here as well. We can have a check in this method to see if the host is empty, if so then don't concat it while creating the policy name

Reply:
oh good catch. thank you

Reply:
Could we generalize the `buildIngressPolicyName` in a separate PR?

Reply:
yes